core/src/main/scala/kafka/coordinator/GroupMetadataManager.scala

```diff
@@ -176,7 +177,7 @@ class GroupMetadataManager(val brokerId: Int,
   def prepareStoreGroup(group: GroupMetadata,
                         groupAssignment: Map[String, Array[Byte]],
                         responseCallback: Errors => Unit): DelayedStore = {
-    val (magicValue, timestamp) = getMessageFormatVersionAndTimestamp(partitionFor(group.groupId))
+    val (magicValue, timestamp) = getMessageFormatVersionAndTimestampIfReplicaIsLocal(partitionFor(group.groupId))
```

Comment:
Seems a bit indirect to propagate the exception all the way up to the `KafkaApis` when we have the callback we need right here. Same for `prepareStoreOffsets` below. I'm wondering if we could have `getMessageFormatVersionAndTimestamp` return an `Option` and treat `None` as the error case? That would be consistent with `ReplicaManager.getMessageFormatVersion`.